### PR TITLE
feat: merge the QGB v0.46.x branch to the default one

### PR DESCRIPTION
**Pull request by SweeXordious**
Merge the QGB feature branch to the default branch in preparation for the QGB app merge.

### CODE REVIEW

File: x/auth/client/testutil/suite.go [hunk 1]
```diff
@@ -644,15 +644,18 @@ func (s *IntegrationTestSuite) TestCLIQueryTxsCmdByEvents() {
 		name        string
 		args        []string
 		expectEmpty bool
+		expectError string
 	}{
-		{
+		{ // Multiple shares failing test
```
Comment (SweeXordious) [inline x/auth/client/testutil/suite.go:649]:
```suggestion
```

File: x/auth/client/testutil/suite.go [hunk 2]
```diff
@@ -644,15 +644,18 @@ func (s *IntegrationTestSuite) TestCLIQueryTxsCmdByEvents() {
...
 			"fee event happy case",
 			[]string{
 				fmt.Sprintf("--events=tx.fee=%s",
 					sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
 				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
 			},
 			false,
+			"transaction spanned more than two shares, this is not yet supported",
+			// TODO: change this to not expect an error when functionality is added to celestia-core
```
Comment (SweeXordious) [inline x/auth/client/testutil/suite.go:658]:
```suggestion
```

File: x/auth/client/testutil/suite.go [hunk 3]
```diff
@@ -662,6 +665,7 @@ func (s *IntegrationTestSuite) TestCLIQueryTxsCmdByEvents() {
 				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
 			},
 			true,
+			"",
```
Comment (SweeXordious) [inline x/auth/client/testutil/suite.go:668]:
```suggestion
```

File: x/staking/simulation/operations.go [hunk 4]
```diff
@@ -158,7 +159,12 @@ func SimulateMsgCreateValidator(ak types.AccountKeeper, bk types.BankKeeper, k k
 			simtypes.RandomDecAmount(r, maxCommission),
 		)
 
-		msg, err := types.NewMsgCreateValidator(address, simAccount.ConsKey.PubKey(), selfDelegation, description, commission, sdk.OneInt())
+		orchAddr := simAccount.Address
+		// create an Ethereum address from the orchestrator address
+		// this is mainly to have a deterministic way of generating an Ethereum address on every run // to have a long enough bytes array
```
Comment (SweeXordious) [inline x/staking/simulation/operations.go:164]:
```suggestion
		// this is mainly to have a deterministic way of generating an Ethereum address on every run
		// to have a long enough bytes array
```

**Review by SweeXordious — COMMENTED** [review]
cosmetics

**Review by evan-forbes — COMMENTED** [review]
LGTM besides the linter ofc

File: proto/cosmos/staking/v1beta1/staking.proto [hunk 5]
```diff
@@ -124,6 +124,12 @@ message Validator {
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
     (gogoproto.nullable)   = false
   ];
+  // The orchestrator field is a celes1... string  (i.e. sdk.AccAddress) that
+  // references the key that is being delegated to
+  string orchestrator = 12;
+  // This is a hex encoded 0x Ethereum public key that will be used by this
+  // validator on Ethereum
+  string eth_address = 13;
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply proto/cosmos/staking/v1beta1/staking.proto:132]: yes, we will need them.

**Comment by SweeXordious** [comment]
@evan-forbes The linter is out of hand. It is failing because of a missing package:
```
Failure: buf.build/googleapis/googleapis:40f07f5b563941f2b20b991a7aedd53d: does not exist
make: *** [Makefile:431: proto-lint] Error 1
```
Probably because we're using `bufbuild/buf:1.0.0-rc8` for generating proto and linting.
We can think of updating the version of `buf` to match upstream:

```
DOCKER_BUF := $(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace bufbuild/buf:1.7.0
```

But that can be handled in a separate PR. Let me know if you agree to create an issue.

PS: I did update it locally, but all the proto generated files were changed. So, it's out of scope of this PR.

For the other failing CI, it's the traditional breaking changes.

Let me know if this is alright, so we merge.

**Comment by evan-forbes** [comment]
Yeah, let's create an issue and update the proto generation.

**Review by evan-forbes — APPROVED** [review]
:+1:

**Comment by SweeXordious** [comment]
@evan-forbes we clicked on the same time :smiley:

**Comment by evan-forbes** [comment]
:muscle:

**Pull request merged**